dj_database_url/__init__.py

```diff
@@ -156,29 +174,56 @@ def parse(
         )
 
     # Pass the query string into OPTIONS.
-    options: Dict[str, Any] = {}
+    query_string_options: Dict[str, Any] = {}
     for key, values in query.items():
         if spliturl.scheme == "mysql" and key == "ssl-ca":
-            options["ssl"] = {"ca": values[-1]}
+            query_string_options["ssl"] = {"ca": values[-1]}
             continue
 
-        options[key] = values[-1]
+        query_string_options[key] = values[-1]
 
     if ssl_require:
-        options["sslmode"] = "require"
+        query_string_options["sslmode"] = "require"
 
     # Support for Postgres Schema URLs
-    if "currentSchema" in options and engine in (
+    if "currentSchema" in query_string_options and engine in (
         "django.contrib.gis.db.backends.postgis",
         "django.db.backends.postgresql_psycopg2",
         "django.db.backends.postgresql",
         "django_redshift_backend",
         "timescale.db.backends.postgresql",
         "timescale.db.backends.postgis",
     ):
-        options["options"] = "-c search_path={0}".format(options.pop("currentSchema"))
+        query_string_options["options"] = "-c search_path={0}".format(
+            query_string_options.pop("currentSchema")
+        )
+
+    if query_string_options:
+        parsed_config["OPTIONS"] = query_string_options
+
+    check_if_query_string_options_overlaps_options(
+        query_string_options=query_string_options, options=options
+    )
 
     if options:
         parsed_config["OPTIONS"] = options
```

Comment:
So, this seems very weird to me as `parsed_config["OPTIONS"]` is being set both on line 209 and 202?